assets/src/edit-story/components/panels/layer/provider.js

```diff
@@ -26,20 +26,85 @@ import StoryPropTypes from '../../../types';
 import Context from './context';
 import useLayers from './useLayers';
 
+const SCROLL_DISTANCE = 5;
+
 function LayerProvider({ children }) {
   const layers = useLayers();
+  const [hasScrollAbove, setHasScrollAbove] = useState(false);
+  const [hasScrollBelow, setHasScrollBelow] = useState(false);
   const [isReordering, setIsReordering] = useState(false);
+  const [scrollDirection, setScrollDirection] = useState(0);
   const [currentSeparator, setCurrentSeparator] = useState(null);
+  const scrollTarget = useRef(null);
+
+  const setScrollTarget = useCallback(
+    (node) => (scrollTarget.current = node.parentElement),
+    []
+  );
+
+  const updateScrollMarkers = useCallback(() => {
+    const node = scrollTarget.current;
+    if (!node) {
+      return;
+    }
+    const possibleScroll = node.scrollHeight - node.clientHeight;
+    if (possibleScroll <= 0) {
+      setHasScrollAbove(false);
+      setHasScrollBelow(false);
+    } else {
+      setHasScrollAbove(node.scrollTop > 0);
+      setHasScrollBelow(node.scrollTop < possibleScroll);
+    }
+  }, []);
+
+  const stopScroll = useCallback(() => setScrollDirection(0), [
+    setScrollDirection,
+  ]);
+
+  const canScrollUp = isReordering && hasScrollAbove;
+  const canScrollDown = isReordering && hasScrollBelow;
+
+  useEffect(() => {
+    const isNotScrolling = scrollDirection === 0;
+    const isScrollingUp = canScrollUp && scrollDirection < 0;
+    const isScrollingDown = canScrollDown && scrollDirection > 0;
+    const isScrollingAnywhere = isScrollingUp || isScrollingDown;
+    if (isNotScrolling || !isScrollingAnywhere) {
+      return undefined;
+    }
+
+    let mounted = true;
+    const update = () => {
+      scrollTarget.current.scrollTop += scrollDirection * SCROLL_DISTANCE;
+      updateScrollMarkers();
+      if (mounted) {
+        window.requestAnimationFrame(update);
```

Comment:
Maybe cancel the last animation frame on dismount?

Reply:
That didn't do the trick alone, unfortunately. Not really sure why, but the loop kept going. With this unmount check, the loop aborts correctly. I felt that doing both was overkill.